ISISScientificComputing/autoreduce
# System test

### Summary of work
* Added utility functions to start and stop the `QueueProcessors` and `activemq service`
* Updated test_settings.py files for various
* Added end to end test for `WISH` and test for `WISH` failure with invalid user script
* Changed autoreduction processor to use the same python binary that is executing the code to run sub processes
  * This is an error we had previously avoided by luck. In the case that a virtual environment is being used to run autoreduction python code, we should not be starting processing using "python" as the interpreter. This is because this will use the default system installation of python not the virtual environment were the code is running as such we can not grantee that autoreduction and it's dependencies are install on system python. 
* Fixed other tests to start and stop activeMQ as required


## How to test your work
**This can only be tested on a linux machine**
### Test the system test locally
* Set up the testing database `LOCALLY` - **DO NOT DROP THE DATABASE ON THE DEV OR PRODUCTION MACHINES**
```
$ mysql -u root 
> database drop autoreduction;
> exit
$ cd path/to/git/repo/for/autoreduction
$ mysql -u root < build/database/test_db_setup.sql
$ python setup.py database
$ mysql -u root < build/database/populate_reduction_viewer.sql
```
* migrate the test_settings
```
python setup.py test_settings
```
* Attempt to run the system tests. Note: these will unless you have sudo access for your current session. 
	* Ensure you have sudo access for the current session
```
$ sudo ls
```
* Attempt to run the tests again - they should pass!


### Test code changes on development environment
* log in to the compute node for the development environment 
```
$ git fetch-p
$ git checkout 28-system-test
$ git pull origin 28-system-test
```
* Once the code has been updated attempt to manually submit some runs using `scripts/manual_submission_script/manual_submission.py`
* Ensure runs successfully complete

Fixes #28
